### ISSUE TITLE
More helpful error message when schema validation fails

### ISSUE THREAD

**Issue by jbusecke** [pangeo-forge/pangeo-forge-runner#160]
Over in https://github.com/leap-stc/data-management/pull/75 i encountered a rather cryptic error:

```
***"message": "Error during running: ", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/bin/pangeo-forge-runner\", line 8, in <module>\n    sys.exit(main())\n  File \"/usr/local/lib/python3.10/dist-packages/pangeo_forge_runner/cli.py\", line 28, in main\n    app.start()\n  File \"/usr/local/lib/python3.10/dist-packages/pangeo_forge_runner/cli.py\", line 23, in start\n    super().start()\n  File \"/usr/local/lib/python3.10/dist-packages/traitlets/config/application.py\", line 475, in start\n    return self.subapp.start()\n  File \"/usr/local/lib/python3.10/dist-packages/pangeo_forge_runner/commands/bake.py\", line 229, in start\n    recipes = feedstock.parse_recipes()\n  File \"/usr/local/lib/python3.10/dist-packages/pangeo_forge_runner/feedstock.py\", line 80, in parse_recipes\n    assert any(\nAssertionError", "status": "failed"***
```

It seemed related to the new [schema validation](#137) but I had to ask @cisaacstern to get to the bottom of the problem. 

Could we raise a more informative error message like:
```
Your recipe_id:`my_recipe` contains fields ['not_this', 'worse'] which are not allowed, only allowed values are ['def_this', 'yay']
```

**Comment by cisaacstern** [comment]
Agreed that the error message should be more descriptive. This error is being raised in feedstock.py, and taking a closer look at this, I now feel I disagree with this comment that I left there: 

https://github.com/pangeo-forge/pangeo-forge-runner/blob/a5816047471df595241a1c57fd4c207e61c6fbb5/pangeo_forge_runner/feedstock.py#L81-L83

This double-validation feels like a code smell to me. The validation should just happen in the MetaYaml schema, and if that validation is leaky, we should fix it there, not double-check it somewhere else (as we are now doing, which is my fault).

So the place this should be fixed is here:

https://github.com/pangeo-forge/pangeo-forge-runner/blob/a5816047471df595241a1c57fd4c207e61c6fbb5/pangeo_forge_runner/meta_yaml.py#L42-L47

Interestingly, this block did not raise an error (i.e. the validation here _is_ currently leaky), which suggests to me that the issue is that we are allowing extra fields in that validation block, which I think we should not be doing.

**Comment by yuvipanda** [comment]
This sounds like the assert caught an issue with the schema validation actually being leaky, so it can be fixed there. Sounds like a win + positive result of the comment you left + assert there :)
